### ISSUE TITLE
ObjectSchema() cannot be applied to properties with a class (instance) type

### ISSUE THREAD

**Issue by laurence-myers** [joiful-ts/joiful#11]
```typescript
class FooClass {
    bar: number;
}

class BazSchema {
    @ObjectConstraints.ObjectSchema()
    qux: FooClass;
}
```

This fails because `ObjectSchema()` only accepts properties of type `Object`, but the metadata value of class instances have a type of `Function`, because in JavaScript classes are syntactic sugar over functions.

A quick fix is to modify the decorator to allow "Function" as a valid type.

**Comment by jbadilla** [comment]
how did you solve this? would you please post a snippet of code?

by the way, great job on this library!! I'm using it for all my TS back-end work. This, I would say, is the only shortcoming.

**Comment by laurence-myers** [comment]
Thanks, I'm glad you find it useful!

I don't have the code in front of me, but I think you can work around it by defining an interface, extending the class from the interface, and using the interface as the property type.

```
interface IFoo {
    bar: number;
}

class FooClass extends IFoo {
    bar: number;
}

class BazSchema {
    @ObjectConstraints.ObjectSchema()
    qux: IFoo;
}
```

I'd welcome a PR to fix this issue, it's a trivial change (but should have a unit test).

**Comment by laurence-myers** [comment]
Fixed in commit c64fce8c282c59b97a6bec39bc579874d0f8def7.

**Comment by laurence-myers** [comment]
Fixed in [v0.0.12](https://github.com/laurence-myers/tsdv-joi/releases/tag/v0.0.12).
